clients/cobol-lsp-vscode-extension/src/extension.ts

```diff
@@ -67,9 +67,10 @@ async function initialize(context: vscode.ExtensionContext) {
     outputChannel.appendLine(message);
     throw Error(message);
   }
-  const copyBooksDownloader = new CopybookDownloadService(
+  const copyBooksDownloader = await CopybookDownloadService.getInstance(
     context.globalStorageUri.fsPath,
   );
```

Comment:
Why not just
```suggestion
  const copyBooksDownloader = new CopybookDownloadService(
    context.globalStorageUri.fsPath,
    await Utils.getZoweExplorerAPI(),
  );
```

and you can also test it more easily.

Reply:
Yes, I thought of making it singleton ( which is not needed , if I think now) and overcomplicated. I will adopt the suggested changes

Reply:
updated